### PR TITLE
awscloud: add import role parameter to AMI registration

### PR DESCRIPTION
**Pull request by ondrejbudai**
We got a request for supporting custom role names, let's start by
adding support for it in the images library.

### CODE REVIEW

File: cmd/boot-aws/main.go [hunk 1]
```diff
@@ -192,6 +192,13 @@ func doSetup(a *awscloud.AWS, filename string, flags *pflag.FlagSet, res *resour
 		return err
 	}
 
+	var importRolePtr *string
+	if importRole, err := flags.GetString("import-role"); importRole != "" {
```
Comment (mvo5) [inline cmd/boot-aws/main.go:196]: (nitpick) Maybe the slightly more canoical:
```go
	var importRolePtr *string
	importRole, err := flags.GetString("import-role")
	if err != nil {
		return err
	}
	if importRole != "" {
		importRolePtr = &importRole
	}
```
ie. error check first? I know it's the same but when reading go code anything that diverges from this pattern automatically stops my flow.

**Review by mvo5 — APPROVED** [review]
Looks good, one tiny style suggestion

**Review by supakeen — APPROVED** [review]

File: cmd/boot-aws/main.go (hunk 1 above)
Comment (achilleas-k) [reply cmd/boot-aws/main.go:196]: I agree with errors first but I do like the limited scoping you get from the `if ret, err := func(); err != nil ...`

Comment (ondrejbudai) [reply cmd/boot-aws/main.go]: What about extracting a helper function to get both error handling first, and limiting the scope? See my latest force-push.

**Review by mvo5 — APPROVED** [review]

File: cmd/boot-aws/main.go (hunk 1 above)
Comment (achilleas-k) [reply cmd/boot-aws/main.go]: Well, Michael approved, so I'm resolving this and merging.

**Review by achilleas-k — APPROVED** [review]
LGTM.  Optional string flag getter is nice.

**Pull request merged**